src/pages/DetailsPage.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import { onUpdated } from 'vue'
+import { getDetails } from '@/helpers/homepageHelpers'
+import { useRoute, useRouter } from 'vue-router'
+import type { DetailedPokemon } from '../helpers/types'
+import Card from 'primevue/card'
+import Carousel from 'primevue/carousel'
+import Image from 'primevue/image'
+import Button from 'primevue/button'
+
+const route = useRoute()
+const router = useRouter()
+const pokeId = route.query.id?.toString()!
+const isshowDetails = ref(false)
+const data = ref<DetailedPokemon>({
+  name: 'bulbasaur',
+  id: 1,
+  types: [
+    { type: { name: 'grass', url: 'https://pokeapi.co/api/v2/type/12/' } },
+    { type: { name: 'poison', url: 'https://pokeapi.co/api/v2/type/4/' } }
+  ],
+  abilities: [
+    { ability: { name: 'overgrow', url: 'https://pokeapi.co/api/v2/ability/65/' } },
+    { ability: { name: 'chlorophyll', url: 'https://pokeapi.co/api/v2/ability/34/' } }
+  ],
+  sprites: {
+    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png'
+  }
+})
+
+const sprites = computed(() => {
+  const outSprites = []
+  for (const key in data.value.sprites) {
+    const sprite = data.value.sprites[key]
+    if (typeof sprite === 'string') {
+      outSprites.push(sprite)
+    }
+  }
+  console.log(outSprites)
+  return outSprites
+})
```

Comment:
Could be rewritten to use map by using
 ```typescript
Object.fromEntries(Object.entries(data.value.sprites).map(([key, value]) => ...))
```
but that might be less clear to some people